inference-engine/src/inference_engine/ie_rtti.cpp

```diff
@@ -97,12 +97,12 @@ template struct InferenceEngine::Parameter::RealData<std::vector<unsigned long>>
 template struct InferenceEngine::Parameter::RealData<std::tuple<unsigned int, unsigned int>>;
 template struct InferenceEngine::Parameter::RealData<std::tuple<unsigned int, unsigned int, unsigned int>>;
 template struct InferenceEngine::Parameter::RealData<InferenceEngine::Blob::Ptr>;
-#endif  // __clang__
+#endif  // __clang__ && !__SYCL_COMPILER_VERSION
```

Comment:
If you need Parameter as well as TBlob so add the same macro to https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/include/ie_parameter.hpp

Have you tested the compilation?

Reply:
Oops. I could build 2021.2 tag version but cmake failed with the current master branch. So, I can't compiled the code now. Sorry to trouble you. Do you have any suggestion for following cmake error (Windows) ?

CMake Error at inference-engine/thirdparty/CMakeLists.txt:98 (add_library):
  add_library cannot create ALIAS target "mkldnn" because target "dnnl" does
  not already exist.

I have checked the cmake file at at inference-engine/thirdparty/CMakeLists.txt:98 but no idea how to resolve the issue.
 98       add_library(mkldnn ALIAS dnnl)

* I have downloaded MKL-DNN with 'prepare_mkl.bat'.
* Also, I have installed oneAPI base kit (w/DNNL) and DNNLROOT environment variable is properly set (c:\Program Files (x86)\Intel\oneAPI\dnnl\2021.1.1\env\..\cpu_tbb).

Reply:
I have tried to build the master branch without CPU plugin (MKL plugin) by adding -DENABLE_MKL_DNN=OFF to the cmake options, and I could build IE without any error (Windows 10).

Do I still need to add the same macro to ie_parameter.hpp for Android?

Reply:
I've added the same compiler check macro to ie_parameter.cpp and pushed it as  the fix #6.

Reply:
Thank you!